### PR TITLE
doc/drivegroups: add docs for DriveGroups with excessive examples

### PR DESCRIPTION
**Pull request by jschmid1**
Please have a look at the structure of this document and provide feedback if needed.

To be done:
- [x] proper link to drivegroup specification docs
- [x] decide where to put it. (install, osd, mgr, bootstrap)
- [x] add it to corresponding toctree

Signed-off-by: Joshua Schmid <jschmid@suse.de>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
What about also moving the content of https://docs.ceph.com/docs/master/mgr/orchestrator_modules/#ceph.deployment.drive_group.DriveGroupSpec into drivegroups.rst?

**Comment by jschmid1** [comment]
> What about also moving the content of https://docs.ceph.com/docs/master/mgr/orchestrator_modules/#ceph.deployment.drive_group.DriveGroupSpec into drivegroups.rst?

it may be too technical for a regular docs file. Let's see how it pans out.

**Comment by jschmid1** [comment]
> /home/jenkins-build/build/workspace/ceph-pr-docs/doc/drivegroups.rst: WARNING: document isn't included in any toctree

It actually is: https://github.com/ceph/ceph/pull/33044/files#diff-0ced3dfd44ba156272e6db34a5247462R96

File: doc/drivegroups.rst [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+* 10 SSDs should share 2 NVMes
+
+This can be described with two layouts.
+
+.. code-block:: yaml
+
+    drive_group_hdd:
+      host_pattern: '*'
+      data_devices:
+        rotational: 0
```
Comment (bk201) [inline doc/drivegroups.rst]:
```suggestion
        rotational: 1
```

File: doc/drivegroups.rst [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        limit: 2 (db_slots is actually to be favoured here, but it's not implemented yet)
+
+    drive_group_ssd:
+      host_pattern: '*'
+      data_devices:
+        model: MC-55-44-XZ
+      db_devices:
+        vendor: VendorC
+
+This would create the desired layout by using all HDDs as data_devices with two SSD assigned as dedicated db/wal devices.
```
Comment (bk201) [inline doc/drivegroups.rst]: Will any WAL devices be created in this example?

File: doc/drivegroups.rst [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+    5 NVMEs
+    Vendor: Intel
+    Model: SSD-123-foo
+    Size: 4TB
+    20 SSDs
+    Vendor: VendorA
+    Model: MC-55-44-ZX
+    Size: 512GB
+
+You can use the 'host_pattern' key in the layout to target certain nodes. Salt target notation helps to keep things easy.
```
Comment (bk201) [inline doc/drivegroups.rst]:
```suggestion
You can use the 'host_pattern' key in the layout to target certain nodes.
```

File: doc/drivegroups.rst [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+------------------
+
+All previous cases co-located the WALs with the DBs.
+It's however possible to deploy the WAL on a dedicated device as well, if it makes sense.
+
+::
+
+    20 HDDs
+    Vendor: VendorA
+    Model: SSD-123-foo
```
Comment (bk201) [inline doc/drivegroups.rst]:
```suggestion
    Model: MC-55-44-ZX
```

File: doc/drivegroups.rst [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+    Size: 4TB
+
+    2 SSDs
+    Vendor: VendorB
+    Model: MC-55-44-ZX
```
Comment (bk201) [inline doc/drivegroups.rst]:
```suggestion
    Model: SSD-123-foo
```
These two model names should be swap to match the paragraph below.

**Comment by sebastian-philipp** [comment]
ping @jschmid1 . Can we get this in?

File: doc/drivegroups.rst (hunk 2 above)
Comment (jschmid1) [reply doc/drivegroups.rst]: Yes, they will be created alongside the `db_devices`

**Comment by jschmid1** [comment]
> ping @jschmid1 . Can we get this in?

I'd like to but I'm struggeling with https://github.com/ceph/ceph/pull/33044#issuecomment-581854976 maybe @zdover23 is the right one for that issue?

File: doc/drivegroups.rst [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+.. code-block:: yaml
+
+    example_drive_group:
+      data_devices:
+        all: true
+      encrypted: true
+
+See a full list in the DriveGroupSpecs
+
+.. py:currentmodule:: orchestrator
```
Comment (sebastian-philipp) [inline doc/drivegroups.rst]:
```suggestion
.. py:currentmodule:: ceph.deployment.drive_group
```

**Comment by sebastian-philipp** [comment]
the only thing left seems to be 
```
/home/jenkins-build/build/workspace/ceph-pr-docs/build-doc/virtualenv/lib/python3.6/site-packages/sphinx/util/docutils.py:285: RemovedInSphinx30Warning: function based directive support is now deprecated. Use class based directive instead.
  RemovedInSphinx30Warning)
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:2: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:9: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.block_db_size, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:16: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.block_wal_size, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:23: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.data_devices, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:30: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.data_directories, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:37: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.db_devices, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:44: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.db_slots, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:51: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.encrypted, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:58: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.host_pattern, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:65: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.journal_devices, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:72: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.journal_size, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:79: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.objectstore, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:86: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.osd_id_claims, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:95: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.osds_per_device, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:103: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.wal_devices, other instance in drivegroups, use :noindex: for one of them
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/mgr/orchestrator_modules.rst:110: WARNING: duplicate object description of ceph.deployment.drive_group.DriveGroupSpec.wal_slots, other instance in drivegroups, use :noindex: for one of them
```

**Comment by jschmid1** [comment]
aand we're back to: 

`/home/jenkins-build/build/workspace/ceph-pr-docs/doc/drivegroups.rst: WARNING: document isn't included in any toctree
`

**Comment by sebastian-philipp** [comment]
plz bring this up in the next stanup.

**Comment by jschmid1** [comment]
rebased atop #https://github.com/ceph/ceph/pull/33551

still `/home/jenkins-build/build/workspace/ceph-pr-docs/doc/drivegroups.rst: WARNING: document isn't included in any toctree`

:100:

**Comment by sebastian-philipp** [comment]
move it into cephadm/ and it will work fine

**Comment by jschmid1** [comment]
> move it into cephadm/ and it will work fine

indeed! thanks for the help

**Comment by sebastian-philipp** [comment]
jenkins render docs

**Comment by jschmid1** [comment]
looks good!

**Pull request merged**